packages/frontend/src/components/SimpleTable/index.tsx

```diff
@@ -43,10 +40,19 @@ const SimpleTable: FC = () => {
         }
         return {};
     }, [explore, resultsData]);
+    const rows = mapDataToTable(tableItems, columnOrder);
+    const headers = columnOrder.map((fieldId) => {
+        const field =
+            explore && fieldId
+                ? findFieldByIdInExplore(explore, fieldId)
+                : undefined;
 
-    const rows = mapDataToTable(tableItems, headers);
-    const validData = rows && headers;
+        return field
+            ? `${field.tableLabel} ${field.label}`
```

Comment:
I don't think this will work with custom metrics and table calculations since they aren't part of the explore. 

we probably need something like this to gather all the possible fields/items:
```
    const dimensionsInMetricQuery = explore
        ? getDimensions(explore).filter((field) =>
              resultsData?.metricQuery.dimensions.includes(fieldId(field)),
          )
        : [];

    const metricsAndTableCalculations: Array<Metric | TableCalculation> =
        explore
            ? [
                  ...getMetrics(explore),
                  ...(resultsData?.metricQuery.additionalMetrics || []).reduce<
                      Metric[]
                  >((acc, additionalMetric) => {
                      const table = explore.tables[additionalMetric.table];
                      if (table) {
                          const metric = convertAdditionalMetric({
                              additionalMetric,
                              table,
                          });
                          return [...acc, metric];
                      }
                      return acc;
                  }, []),
                  ...(resultsData?.metricQuery.tableCalculations || []),
              ].filter((item) => {
                  if (isField(item)) {
                      return resultsData?.metricQuery.metrics.includes(
                          fieldId(item),
                      );
                  }
                  return true;
              })
            : [];

    const items = [...dimensionsInMetricQuery, ...metricsAndTableCalculations];
    
```

We can also use `getItemLabel` util function that has some logic to get the correct label.

Reply:
I'm not sure what's the problem with table calculations or custom metrics. Because those are not fields, they will be using the friendlyName specified by the column order , like before. 

![image](https://user-images.githubusercontent.com/1983672/170453764-267ccd8d-ef7c-4375-ad9f-1a63f55ec076.png)

I agree about getItemLabel, I didn't know about that method. I'll update it

Reply:
In that screenshot you can see that the custom metric has a different label. Currently the table calculation works using the friendlyLabel() because how we generate the id/label but this will likely change in the future. `friendlyLabel()` should be used as the fallback.

Reply:
![Screenshot from 2022-05-26 12-12-05](https://user-images.githubusercontent.com/1983672/170468071-50804f31-cd66-425e-8ca2-8d74b7c4906d.png)